gradio/routes.py

```diff
@@ -509,6 +511,27 @@ async def join_queue(
         async def get_queue_status():
             return app.get_blocks()._queue.get_estimation()
 
+        @app.post("/upload", dependencies=[Depends(login_check)])
+        async def upload_file(
+            files: List[UploadFile] = File(...),
+        ):
+            output_files = []
+            for input_file in files:
+                if "." in input_file.filename:
+                    prefix, suffix = input_file.filename.rsplit(".", 1)
+                else:
+                    prefix = input_file.filename
+                    suffix = ""
+                output_file_obj = tempfile.NamedTemporaryFile(delete=False, suffix=f".{suffix}", prefix=f"{prefix}_")
```

Comment:
Nit: if `suffix` is `""`, this will add a `.` to the end of the temporary filename

Note, there might also be issues with the filename containing invalid characters, etc. I would reuse the implementation of `TempFileManager.get_prefix_and_extension()` (maybe make that a classmethod) where we have already handled this logic